OutOfMemoryError on using primitive array on parameterizedTest

## Overview

- **Bug report.** OutOfMemoryError when running parameter based test

```
[DEBUG] Dependency collection stats: {ConflictMarker.analyzeTime=4, ConflictMarker.markTime=0, ConflictMarker.nodeCount=128, ConflictIdSorter.graphTime=1, ConflictIdSorter.topsortTime=0, ConflictIdSorter.conflictIdCount=43, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=7, ConflictResolver.conflictItemCount=94, DefaultDependencyCollector.collectTime=78, DefaultDependencyCollector.transformTime=12}
[DEBUG] org.apache.maven.plugins:maven-surefire-plugin:jar:2.20:
[DEBUG]    org.junit.platform:junit-platform-surefire-provider:jar:1.0.0-M5:runtime
[DEBUG]       org.junit.platform:junit-platform-launcher:jar:1.0.0-M5:runtime
[DEBUG]       org.apache.maven.surefire:common-java5:jar:2.19.1:runtime
[DEBUG]    org.junit.jupiter:junit-jupiter-engine:jar:5.0.0-M5:runtime
[DEBUG]       org.junit.platform:junit-platform-engine:jar:1.0.0-M5:runtime
[DEBUG]          org.junit.platform:junit-platform-commons:jar:1.0.0-M5:runtime
[DEBUG]          org.opentest4j:opentest4j:jar:1.0.0-M2:runtime
[DEBUG]       org.junit.jupiter:junit-jupiter-api:jar:5.0.0-M5:runtime
[DEBUG]    org.apache.maven:maven-plugin-api:jar:2.2.1:compile
[DEBUG]    org.apache.maven.surefire:maven-surefire-common:jar:2.20:compile
[DEBUG]       org.apache.maven.surefire:surefire-booter:jar:2.20:compile
[DEBUG]       org.apache.maven:maven-artifact:jar:2.2.1:compile
[DEBUG]          org.codehaus.plexus:plexus-utils:jar:1.5.15:compile
[DEBUG]       org.apache.maven:maven-plugin-descriptor:jar:2.2.1:compile
[DEBUG]          org.codehaus.plexus:plexus-container-default:jar:1.0-alpha-9-stable-1:compile
[DEBUG]             junit:junit:jar:4.12:compile
[DEBUG]                org.hamcrest:hamcrest-core:jar:1.3:compile
[DEBUG]       org.apache.maven:maven-project:jar:2.2.1:compile
[DEBUG]          org.apache.maven:maven-settings:jar:2.2.1:compile
[DEBUG]          org.apache.maven:maven-profile:jar:2.2.1:compile
[DEBUG]          org.apache.maven:maven-artifact-manager:jar:2.2.1:compile
[DEBUG]             backport-util-concurrent:backport-util-concurrent:jar:3.1:compile
[DEBUG]          org.apache.maven:maven-plugin-registry:jar:2.2.1:compile
[DEBUG]          org.codehaus.plexus:plexus-interpolation:jar:1.11:compile
[DEBUG]       org.apache.maven:maven-model:jar:2.2.1:compile
[DEBUG]       org.apache.maven:maven-core:jar:2.2.1:compile
[DEBUG]          org.apache.maven:maven-plugin-parameter-documenter:jar:2.2.1:compile
[DEBUG]          org.slf4j:slf4j-jdk14:jar:1.5.6:runtime
[DEBUG]             org.slf4j:slf4j-api:jar:1.5.6:runtime
[DEBUG]          org.slf4j:jcl-over-slf4j:jar:1.5.6:runtime
[DEBUG]          org.apache.maven.reporting:maven-reporting-api:jar:3.0:compile
[DEBUG]          org.apache.maven:maven-repository-metadata:jar:2.2.1:compile
[DEBUG]          org.apache.maven:maven-error-diagnostics:jar:2.2.1:compile
[DEBUG]          org.apache.maven:maven-monitor:jar:2.2.1:compile
[DEBUG]          classworlds:classworlds:jar:1.1:compile
[DEBUG]          org.sonatype.plexus:plexus-sec-dispatcher:jar:1.3:compile
[DEBUG]             org.sonatype.plexus:plexus-cipher:jar:1.4:compile
[DEBUG]       org.apache.commons:commons-lang3:jar:3.1:compile
[DEBUG]    org.apache.maven.surefire:surefire-api:jar:2.20:compile
[DEBUG]       org.apache.maven.surefire:surefire-logger-api:jar:2.20:compile
[DEBUG]    org.apache.maven:maven-toolchain:jar:2.2.1:compile
[DEBUG]    org.apache.maven.plugin-tools:maven-plugin-annotations:jar:3.3:compile
[DEBUG] Created new class realm plugin>org.apache.maven.plugins:maven-surefire-plugin:2.20
[DEBUG] Importing foreign packages into class realm plugin>org.apache.maven.plugins:maven-surefire-plugin:2.20
[DEBUG]   Imported:  < maven.api
[DEBUG] Populating class realm plugin>org.apache.maven.plugins:maven-surefire-plugin:2.20
[DEBUG]   Included: org.apache.maven.plugins:maven-surefire-plugin:jar:2.20
[DEBUG]   Included: org.junit.platform:junit-platform-surefire-provider:jar:1.0.0-M5
[DEBUG]   Included: org.junit.platform:junit-platform-launcher:jar:1.0.0-M5
[DEBUG]   Included: org.apache.maven.surefire:common-java5:jar:2.19.1
[DEBUG]   Included: org.junit.jupiter:junit-jupiter-engine:jar:5.0.0-M5
[DEBUG]   Included: org.junit.platform:junit-platform-engine:jar:1.0.0-M5
[DEBUG]   Included: org.junit.platform:junit-platform-commons:jar:1.0.0-M5
[DEBUG]   Included: org.opentest4j:opentest4j:jar:1.0.0-M2
[DEBUG]   Included: org.junit.jupiter:junit-jupiter-api:jar:5.0.0-M5
[DEBUG]   Included: org.apache.maven.surefire:maven-surefire-common:jar:2.20
[DEBUG]   Included: org.apache.maven.surefire:surefire-booter:jar:2.20
[DEBUG]   Included: org.codehaus.plexus:plexus-utils:jar:1.5.15
[DEBUG]   Included: junit:junit:jar:4.12
[DEBUG]   Included: org.hamcrest:hamcrest-core:jar:1.3
[DEBUG]   Included: backport-util-concurrent:backport-util-concurrent:jar:3.1
[DEBUG]   Included: org.codehaus.plexus:plexus-interpolation:jar:1.11
[DEBUG]   Included: org.slf4j:slf4j-jdk14:jar:1.5.6
[DEBUG]   Included: org.slf4j:jcl-over-slf4j:jar:1.5.6
[DEBUG]   Included: org.apache.maven.reporting:maven-reporting-api:jar:3.0
[DEBUG]   Included: org.sonatype.plexus:plexus-sec-dispatcher:jar:1.3
[DEBUG]   Included: org.sonatype.plexus:plexus-cipher:jar:1.4
[DEBUG]   Included: org.apache.commons:commons-lang3:jar:3.1
[DEBUG]   Included: org.apache.maven.surefire:surefire-api:jar:2.20
[DEBUG]   Included: org.apache.maven.surefire:surefire-logger-api:jar:2.20
[DEBUG]   Included: org.apache.maven.plugin-tools:maven-plugin-annotations:jar:3.3
[DEBUG] Configuring mojo org.apache.maven.plugins:maven-surefire-plugin:2.20:test from plugin realm ClassRealm[plugin>org.apache.maven.plugins:maven-surefire-plugin:2.20, parent: sun.misc.Launcher$AppClassLoader@42a57993]
[DEBUG] Configuring mojo 'org.apache.maven.plugins:maven-surefire-plugin:2.20:test' with basic configurator -->
[DEBUG]   (s) additionalClasspathElements = []
[DEBUG]   (s) basedir = /Users/manavdeepbrar/Documents/Coursera_algorithmic_toolbox/primitive.array
[DEBUG]   (s) childDelegation = false
[DEBUG]   (s) classesDirectory = /Users/manavdeepbrar/Documents/Coursera_algorithmic_toolbox/primitive.array/target/classes
[DEBUG]   (s) classpathDependencyExcludes = []
[DEBUG]   (s) dependenciesToScan = []
[DEBUG]   (s) disableXmlReport = false
[DEBUG]   (s) enableAssertions = true
[DEBUG]   (f) forkCount = 1
[DEBUG]   (s) forkMode = once
[DEBUG]   (s) forkedProcessExitTimeoutInSeconds = 30
[DEBUG]   (s) junitArtifactName = junit:junit
[DEBUG]   (s) localRepository =       id: local
      url: file:///Users/manavdeepbrar/.m2/repository/
   layout: default
snapshots: [enabled => true, update => always]
 releases: [enabled => true, update => always]

[DEBUG]   (f) parallelMavenExecution = false
[DEBUG]   (s) parallelOptimized = true
[DEBUG]   (s) perCoreThreadCount = true
[DEBUG]   (s) pluginArtifactMap = {org.apache.maven.plugins:maven-surefire-plugin=org.apache.maven.plugins:maven-surefire-plugin:maven-plugin:2.20:, org.junit.platform:junit-platform-surefire-provider=org.junit.platform:junit-platform-surefire-provider:jar:1.0.0-M5:runtime, org.junit.platform:junit-platform-launcher=org.junit.platform:junit-platform-launcher:jar:1.0.0-M5:runtime, org.apache.maven.surefire:common-java5=org.apache.maven.surefire:common-java5:jar:2.19.1:runtime, org.junit.jupiter:junit-jupiter-engine=org.junit.jupiter:junit-jupiter-engine:jar:5.0.0-M5:runtime, org.junit.platform:junit-platform-engine=org.junit.platform:junit-platform-engine:jar:1.0.0-M5:runtime, org.junit.platform:junit-platform-commons=org.junit.platform:junit-platform-commons:jar:1.0.0-M5:runtime, org.opentest4j:opentest4j=org.opentest4j:opentest4j:jar:1.0.0-M2:runtime, org.junit.jupiter:junit-jupiter-api=org.junit.jupiter:junit-jupiter-api:jar:5.0.0-M5:runtime, org.apache.maven:maven-plugin-api=org.apache.maven:maven-plugin-api:jar:2.2.1:compile, org.apache.maven.surefire:maven-surefire-common=org.apache.maven.surefire:maven-surefire-common:jar:2.20:compile, org.apache.maven.surefire:surefire-booter=org.apache.maven.surefire:surefire-booter:jar:2.20:compile, org.apache.maven:maven-artifact=org.apache.maven:maven-artifact:jar:2.2.1:compile, org.codehaus.plexus:plexus-utils=org.codehaus.plexus:plexus-utils:jar:1.5.15:compile, org.apache.maven:maven-plugin-descriptor=org.apache.maven:maven-plugin-descriptor:jar:2.2.1:compile, org.codehaus.plexus:plexus-container-default=org.codehaus.plexus:plexus-container-default:jar:1.0-alpha-9-stable-1:compile, junit:junit=junit:junit:jar:4.12:compile, org.hamcrest:hamcrest-core=org.hamcrest:hamcrest-core:jar:1.3:compile, org.apache.maven:maven-project=org.apache.maven:maven-project:jar:2.2.1:compile, org.apache.maven:maven-settings=org.apache.maven:maven-settings:jar:2.2.1:compile, org.apache.maven:maven-profile=org.apache.maven:maven-profile:jar:2.2.1:compile, org.apache.maven:maven-artifact-manager=org.apache.maven:maven-artifact-manager:jar:2.2.1:compile, backport-util-concurrent:backport-util-concurrent=backport-util-concurrent:backport-util-concurrent:jar:3.1:compile, org.apache.maven:maven-plugin-registry=org.apache.maven:maven-plugin-registry:jar:2.2.1:compile, org.codehaus.plexus:plexus-interpolation=org.codehaus.plexus:plexus-interpolation:jar:1.11:compile, org.apache.maven:maven-model=org.apache.maven:maven-model:jar:2.2.1:compile, org.apache.maven:maven-core=org.apache.maven:maven-core:jar:2.2.1:compile, org.apache.maven:maven-plugin-parameter-documenter=org.apache.maven:maven-plugin-parameter-documenter:jar:2.2.1:compile, org.slf4j:slf4j-jdk14=org.slf4j:slf4j-jdk14:jar:1.5.6:runtime, org.slf4j:slf4j-api=org.slf4j:slf4j-api:jar:1.5.6:runtime, org.slf4j:jcl-over-slf4j=org.slf4j:jcl-over-slf4j:jar:1.5.6:runtime, org.apache.maven.reporting:maven-reporting-api=org.apache.maven.reporting:maven-reporting-api:jar:3.0:compile, org.apache.maven:maven-repository-metadata=org.apache.maven:maven-repository-metadata:jar:2.2.1:compile, org.apache.maven:maven-error-diagnostics=org.apache.maven:maven-error-diagnostics:jar:2.2.1:compile, org.apache.maven:maven-monitor=org.apache.maven:maven-monitor:jar:2.2.1:compile, classworlds:classworlds=classworlds:classworlds:jar:1.1:compile, org.sonatype.plexus:plexus-sec-dispatcher=org.sonatype.plexus:plexus-sec-dispatcher:jar:1.3:compile, org.sonatype.plexus:plexus-cipher=org.sonatype.plexus:plexus-cipher:jar:1.4:compile, org.apache.commons:commons-lang3=org.apache.commons:commons-lang3:jar:3.1:compile, org.apache.maven.surefire:surefire-api=org.apache.maven.surefire:surefire-api:jar:2.20:compile, org.apache.maven.surefire:surefire-logger-api=org.apache.maven.surefire:surefire-logger-api:jar:2.20:compile, org.apache.maven:maven-toolchain=org.apache.maven:maven-toolchain:jar:2.2.1:compile, org.apache.maven.plugin-tools:maven-plugin-annotations=org.apache.maven.plugin-tools:maven-plugin-annotations:jar:3.3:compile}
[DEBUG]   (f) pluginDescriptor = Component Descriptor: role: 'org.apache.maven.plugin.Mojo', implementation: 'org.apache.maven.plugin.surefire.HelpMojo', role hint: 'org.apache.maven.plugins:maven-surefire-plugin:2.20:help'
role: 'org.apache.maven.plugin.Mojo', implementation: 'org.apache.maven.plugin.surefire.SurefirePlugin', role hint: 'org.apache.maven.plugins:maven-surefire-plugin:2.20:test'
---
[DEBUG]   (s) printSummary = true
[DEBUG]   (s) projectArtifactMap = {org.junit.jupiter:junit-jupiter-api=org.junit.jupiter:junit-jupiter-api:jar:5.0.0-M5:test, org.opentest4j:opentest4j=org.opentest4j:opentest4j:jar:1.0.0-M2:test, org.junit.platform:junit-platform-commons=org.junit.platform:junit-platform-commons:jar:1.0.0-M5:test, org.junit.jupiter:junit-jupiter-params=org.junit.jupiter:junit-jupiter-params:jar:5.0.0-M5:test, org.junit.platform:junit-platform-runner=org.junit.platform:junit-platform-runner:jar:1.0.0-M5:test, org.junit.platform:junit-platform-launcher=org.junit.platform:junit-platform-launcher:jar:1.0.0-M5:test, org.junit.platform:junit-platform-engine=org.junit.platform:junit-platform-engine:jar:1.0.0-M5:test, org.junit.platform:junit-platform-suite-api=org.junit.platform:junit-platform-suite-api:jar:1.0.0-M5:test, junit:junit=junit:junit:jar:4.12:test, org.hamcrest:hamcrest-core=org.hamcrest:hamcrest-core:jar:1.3:test}
[DEBUG]   (s) projectBuildDirectory = /Users/manavdeepbrar/Documents/Coursera_algorithmic_toolbox/primitive.array/target
[DEBUG]   (s) redirectTestOutputToFile = false
[DEBUG]   (s) remoteRepositories = [      id: central
      url: https://repo.maven.apache.org/maven2
   layout: default
snapshots: [enabled => false, update => daily]
 releases: [enabled => true, update => never]
]
[DEBUG]   (s) reportFormat = brief
[DEBUG]   (s) reportsDirectory = /Users/manavdeepbrar/Documents/Coursera_algorithmic_toolbox/primitive.array/target/surefire-reports
[DEBUG]   (f) rerunFailingTestsCount = 0
[DEBUG]   (f) reuseForks = true
[DEBUG]   (s) runOrder = filesystem
[DEBUG]   (f) shutdown = testset
[DEBUG]   (s) skip = false
[DEBUG]   (f) skipAfterFailureCount = 0
[DEBUG]   (s) skipTests = false
[DEBUG]   (s) suiteXmlFiles = []
[DEBUG]   (s) tempDir = surefire
[DEBUG]   (s) testClassesDirectory = /Users/manavdeepbrar/Documents/Coursera_algorithmic_toolbox/primitive.array/target/test-classes
[DEBUG]   (s) testFailureIgnore = false
[DEBUG]   (s) testNGArtifactName = org.testng:testng
[DEBUG]   (s) testSourceDirectory = /Users/manavdeepbrar/Documents/Coursera_algorithmic_toolbox/primitive.array/src/test/java
[DEBUG]   (s) threadCountClasses = 0
[DEBUG]   (s) threadCountMethods = 0
[DEBUG]   (s) threadCountSuites = 0
[DEBUG]   (s) trimStackTrace = true
[DEBUG]   (s) useFile = true
[DEBUG]   (s) useManifestOnlyJar = true
[DEBUG]   (s) useSystemClassLoader = true
[DEBUG]   (s) useUnlimitedThreads = false
[DEBUG]   (s) workingDirectory = /Users/manavdeepbrar/Documents/Coursera_algorithmic_toolbox/primitive.array
[DEBUG]   (s) project = MavenProject: com.gmail.manavbrar.pmitivearray:primitive.array:0.0.1-SNAPSHOT @ /Users/manavdeepbrar/Documents/Coursera_algorithmic_toolbox/primitive.array/pom.xml
[DEBUG]   (s) session = org.apache.maven.execution.MavenSession@4e76dac
[DEBUG] -- end configuration --
[DEBUG] Surefire report directory: /Users/manavdeepbrar/Documents/Coursera_algorithmic_toolbox/primitive.array/target/surefire-reports
[DEBUG] Using configured provider org.junit.platform.surefire.provider.JUnitPlatformProvider
[DEBUG] Setting system property [user.dir]=[/Users/manavdeepbrar/Documents/Coursera_algorithmic_toolbox/primitive.array]
[DEBUG] Setting system property [localRepository]=[/Users/manavdeepbrar/.m2/repository]
[DEBUG] Setting system property [basedir]=[/Users/manavdeepbrar/Documents/Coursera_algorithmic_toolbox/primitive.array]
[DEBUG] parallel='none', perCoreThreadCount=true, threadCount=0, useUnlimitedThreads=false, threadCountSuites=0, threadCountClasses=0, threadCountMethods=0, parallelOptimized=true
[DEBUG] dummy:dummy:jar:1.0 (selected for null)
[DEBUG]   org.apache.maven.surefire:surefire-booter:jar:2.20:compile (selected for compile)
[DEBUG]     org.apache.maven.surefire:surefire-api:jar:2.20:compile (selected for compile)
[DEBUG]       org.apache.maven.surefire:surefire-logger-api:jar:2.20:compile (selected for compile)
[DEBUG] Adding to surefire booter test classpath: /Users/manavdeepbrar/.m2/repository/org/apache/maven/surefire/surefire-booter/2.20/surefire-booter-2.20.jar Scope: compile
[DEBUG] Adding to surefire booter test classpath: /Users/manavdeepbrar/.m2/repository/org/apache/maven/surefire/surefire-api/2.20/surefire-api-2.20.jar Scope: compile
[DEBUG] Adding to surefire booter test classpath: /Users/manavdeepbrar/.m2/repository/org/apache/maven/surefire/surefire-logger-api/2.20/surefire-logger-api-2.20.jar Scope: compile
[DEBUG] Using JVM: /Library/Java/JavaVirtualMachines/jdk1.8.0_112.jdk/Contents/Home/jre/bin/java
[DEBUG] dummy:dummy:jar:1.0 (selected for null)
[DEBUG]   org.apache.maven.plugins:maven-surefire-plugin:maven-plugin:2.20: (selected for )
[DEBUG]     org.apache.maven:maven-plugin-api:jar:2.2.1:runtime (selected for runtime)
[DEBUG]     org.apache.maven.surefire:maven-surefire-common:jar:2.20:runtime (selected for runtime)
[DEBUG]       org.apache.maven.plugin-tools:maven-plugin-annotations:jar:3.3:runtime (selected for runtime)
[DEBUG]         org.apache.maven:maven-artifact:jar:3.0:runtime (selected for runtime)
[DEBUG]           org.codehaus.plexus:plexus-utils:jar:2.0.4:runtime (selected for runtime)
[DEBUG]       org.apache.maven.surefire:surefire-api:jar:2.20:runtime (selected for runtime)
[DEBUG]         org.apache.maven.surefire:surefire-logger-api:jar:2.20:runtime (selected for runtime)
[DEBUG]       org.apache.maven.surefire:surefire-booter:jar:2.20:runtime (selected for runtime)
[DEBUG]       org.apache.maven:maven-artifact:jar:3.0:runtime (removed - nearer found: 2.2.1)
[DEBUG]       org.apache.maven:maven-artifact:jar:2.2.1:runtime (selected for runtime)
[DEBUG]         org.codehaus.plexus:plexus-utils:jar:1.5.15:runtime (selected for runtime)
[DEBUG]       org.apache.maven:maven-plugin-descriptor:jar:2.2.1:runtime (selected for runtime)
[DEBUG]         org.codehaus.plexus:plexus-container-default:jar:1.0-alpha-9-stable-1:runtime (selected for runtime)
[DEBUG]           junit:junit:jar:3.8.1:runtime (selected for runtime)
[DEBUG]           org.codehaus.plexus:plexus-utils:jar:1.0.4:runtime (removed - nearer found: 1.5.15)
[DEBUG]           classworlds:classworlds:jar:1.1-alpha-2:runtime (selected for runtime)
[DEBUG]       org.apache.maven:maven-project:jar:2.2.1:runtime (selected for runtime)
[DEBUG]         org.apache.maven:maven-settings:jar:2.2.1:runtime (selected for runtime)
[DEBUG]           org.apache.maven:maven-model:jar:2.2.1:runtime (selected for runtime)
[DEBUG]           org.codehaus.plexus:plexus-interpolation:jar:1.11:runtime (selected for runtime)
[DEBUG]         org.apache.maven:maven-profile:jar:2.2.1:runtime (selected for runtime)
[DEBUG]         org.apache.maven:maven-model:jar:2.2.1:runtime (selected for runtime)
[DEBUG]         org.apache.maven:maven-artifact-manager:jar:2.2.1:runtime (selected for runtime)
[DEBUG]           org.apache.maven:maven-repository-metadata:jar:2.2.1:runtime (selected for runtime)
[DEBUG]           org.apache.maven.wagon:wagon-provider-api:jar:1.0-beta-6:runtime (selected for runtime)
[DEBUG]             org.codehaus.plexus:plexus-utils:jar:1.4.2:runtime (removed - nearer found: 1.5.15)
[DEBUG]           backport-util-concurrent:backport-util-concurrent:jar:3.1:runtime (selected for runtime)
[DEBUG]         org.apache.maven:maven-plugin-registry:jar:2.2.1:runtime (selected for runtime)
[DEBUG]         org.codehaus.plexus:plexus-interpolation:jar:1.11:runtime (selected for runtime)
[DEBUG]       org.apache.maven:maven-model:jar:2.2.1:runtime (selected for runtime)
[DEBUG]       org.apache.maven:maven-core:jar:2.2.1:runtime (selected for runtime)
[DEBUG]         org.apache.maven:maven-plugin-parameter-documenter:jar:2.2.1:runtime (selected for runtime)
[DEBUG]         org.apache.maven.wagon:wagon-http:jar:1.0-beta-6:runtime (selected for runtime)
[DEBUG]           org.apache.maven.wagon:wagon-http-shared:jar:1.0-beta-6:runtime (selected for runtime)
[DEBUG]             nekohtml:xercesMinimal:jar:1.9.6.2:runtime (selected for runtime)
[DEBUG]             nekohtml:nekohtml:jar:1.9.6.2:runtime (selected for runtime)
[DEBUG]             commons-httpclient:commons-httpclient:jar:3.1:runtime (selected for runtime)
[DEBUG]               commons-codec:commons-codec:jar:1.2:runtime (selected for runtime)
[DEBUG]         org.apache.maven.wagon:wagon-webdav-jackrabbit:jar:1.0-beta-6:runtime (selected for runtime)
[DEBUG]           org.apache.jackrabbit:jackrabbit-webdav:jar:1.5.0:runtime (selected for runtime)
[DEBUG]             org.apache.jackrabbit:jackrabbit-jcr-commons:jar:1.5.0:runtime (selected for runtime)
[DEBUG]             org.slf4j:slf4j-api:jar:1.5.3:runtime (selected for runtime)
[DEBUG]             commons-httpclient:commons-httpclient:jar:3.0:runtime (removed - nearer found: 3.1)
[DEBUG]           org.slf4j:slf4j-nop:jar:1.5.3:runtime (selected for runtime)
[DEBUG]         org.slf4j:slf4j-jdk14:jar:1.5.6:runtime (selected for runtime)
[DEBUG]           org.slf4j:slf4j-api:jar:1.5.3:runtime (removed - nearer found: 1.5.6)
[DEBUG]           org.slf4j:slf4j-api:jar:1.5.6:runtime (selected for runtime)
[DEBUG]         org.slf4j:jcl-over-slf4j:jar:1.5.6:runtime (selected for runtime)
[DEBUG]         org.apache.maven.reporting:maven-reporting-api:jar:2.2.1:runtime (selected for runtime)
[DEBUG]           org.apache.maven.doxia:doxia-logging-api:jar:1.1:runtime (selected for runtime)
[DEBUG]             org.codehaus.plexus:plexus-container-default:jar:1.0-alpha-30:runtime (removed - nearer found: 1.0-alpha-9-stable-1)
[DEBUG]         org.apache.maven.wagon:wagon-provider-api:jar:1.0-beta-6:runtime (selected for runtime)
[DEBUG]           org.codehaus.plexus:plexus-utils:jar:1.4.2:runtime (removed - nearer found: 1.5.15)
[DEBUG]         org.apache.maven:maven-repository-metadata:jar:2.2.1:runtime (selected for runtime)
[DEBUG]         org.apache.maven:maven-error-diagnostics:jar:2.2.1:runtime (selected for runtime)
[DEBUG]         org.apache.maven:maven-monitor:jar:2.2.1:runtime (selected for runtime)
[DEBUG]         classworlds:classworlds:jar:1.1-alpha-2:runtime (removed - nearer found: 1.1)
[DEBUG]         classworlds:classworlds:jar:1.1:runtime (selected for runtime)
[DEBUG]         org.sonatype.plexus:plexus-sec-dispatcher:jar:1.3:runtime (selected for runtime)
[DEBUG]           org.codehaus.plexus:plexus-utils:jar:1.5.5:runtime (removed - nearer found: 1.5.15)
[DEBUG]           org.sonatype.plexus:plexus-cipher:jar:1.4:runtime (selected for runtime)
[DEBUG]       org.apache.maven:maven-toolchain:jar:2.2.1:runtime (selected for runtime)
[DEBUG]       org.apache.commons:commons-lang3:jar:3.1:runtime (selected for runtime)
[DEBUG]     org.apache.maven.surefire:surefire-api:jar:2.20:runtime (selected for runtime)
[DEBUG]       org.apache.maven.surefire:surefire-logger-api:jar:2.20:runtime (selected for runtime)
[DEBUG]     org.apache.maven:maven-toolchain:jar:2.2.1:runtime (selected for runtime)
[DEBUG]     org.apache.maven.plugin-tools:maven-plugin-annotations:jar:3.3:runtime (selected for runtime)
[DEBUG]       org.apache.maven:maven-artifact:jar:3.0:runtime (removed - nearer found: 2.2.1)
[DEBUG] test classpath:  /Users/manavdeepbrar/Documents/Coursera_algorithmic_toolbox/primitive.array/target/test-classes  /Users/manavdeepbrar/Documents/Coursera_algorithmic_toolbox/primitive.array/target/classes  /Users/manavdeepbrar/.m2/repository/org/junit/jupiter/junit-jupiter-api/5.0.0-M5/junit-jupiter-api-5.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/org/opentest4j/opentest4j/1.0.0-M2/opentest4j-1.0.0-M2.jar  /Users/manavdeepbrar/.m2/repository/org/junit/platform/junit-platform-commons/1.0.0-M5/junit-platform-commons-1.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/org/junit/jupiter/junit-jupiter-params/5.0.0-M5/junit-jupiter-params-5.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/org/junit/platform/junit-platform-runner/1.0.0-M5/junit-platform-runner-1.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/org/junit/platform/junit-platform-launcher/1.0.0-M5/junit-platform-launcher-1.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/org/junit/platform/junit-platform-engine/1.0.0-M5/junit-platform-engine-1.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/org/junit/platform/junit-platform-suite-api/1.0.0-M5/junit-platform-suite-api-1.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/junit/junit/4.12/junit-4.12.jar  /Users/manavdeepbrar/.m2/repository/org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar
[DEBUG] provider classpath:  /Users/manavdeepbrar/.m2/repository/org/junit/platform/junit-platform-surefire-provider/1.0.0-M5/junit-platform-surefire-provider-1.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/org/junit/platform/junit-platform-launcher/1.0.0-M5/junit-platform-launcher-1.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/org/apache/maven/surefire/common-java5/2.19.1/common-java5-2.19.1.jar  /Users/manavdeepbrar/.m2/repository/org/junit/jupiter/junit-jupiter-engine/5.0.0-M5/junit-jupiter-engine-5.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/org/junit/platform/junit-platform-engine/1.0.0-M5/junit-platform-engine-1.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/org/junit/platform/junit-platform-commons/1.0.0-M5/junit-platform-commons-1.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/org/opentest4j/opentest4j/1.0.0-M2/opentest4j-1.0.0-M2.jar  /Users/manavdeepbrar/.m2/repository/org/junit/jupiter/junit-jupiter-api/5.0.0-M5/junit-jupiter-api-5.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/junit/junit/4.12/junit-4.12.jar  /Users/manavdeepbrar/.m2/repository/org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar  /Users/manavdeepbrar/.m2/repository/org/apache/maven/reporting/maven-reporting-api/3.0/maven-reporting-api-3.0.jar
[DEBUG] test(compact) classpath:  test-classes  classes  junit-jupiter-api-5.0.0-M5.jar  opentest4j-1.0.0-M2.jar  junit-platform-commons-1.0.0-M5.jar  junit-jupiter-params-5.0.0-M5.jar  junit-platform-runner-1.0.0-M5.jar  junit-platform-launcher-1.0.0-M5.jar  junit-platform-engine-1.0.0-M5.jar  junit-platform-suite-api-1.0.0-M5.jar  junit-4.12.jar  hamcrest-core-1.3.jar
[DEBUG] provider(compact) classpath:  junit-platform-surefire-provider-1.0.0-M5.jar  junit-platform-launcher-1.0.0-M5.jar  common-java5-2.19.1.jar  junit-jupiter-engine-5.0.0-M5.jar  junit-platform-engine-1.0.0-M5.jar  junit-platform-commons-1.0.0-M5.jar  opentest4j-1.0.0-M2.jar  junit-jupiter-api-5.0.0-M5.jar  junit-4.12.jar  hamcrest-core-1.3.jar  maven-reporting-api-3.0.jar
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[DEBUG] boot classpath:  /Users/manavdeepbrar/.m2/repository/org/apache/maven/surefire/surefire-booter/2.20/surefire-booter-2.20.jar  /Users/manavdeepbrar/.m2/repository/org/apache/maven/surefire/surefire-api/2.20/surefire-api-2.20.jar  /Users/manavdeepbrar/.m2/repository/org/apache/maven/surefire/surefire-logger-api/2.20/surefire-logger-api-2.20.jar  /Users/manavdeepbrar/Documents/Coursera_algorithmic_toolbox/primitive.array/target/test-classes  /Users/manavdeepbrar/Documents/Coursera_algorithmic_toolbox/primitive.array/target/classes  /Users/manavdeepbrar/.m2/repository/org/junit/jupiter/junit-jupiter-api/5.0.0-M5/junit-jupiter-api-5.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/org/opentest4j/opentest4j/1.0.0-M2/opentest4j-1.0.0-M2.jar  /Users/manavdeepbrar/.m2/repository/org/junit/platform/junit-platform-commons/1.0.0-M5/junit-platform-commons-1.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/org/junit/jupiter/junit-jupiter-params/5.0.0-M5/junit-jupiter-params-5.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/org/junit/platform/junit-platform-runner/1.0.0-M5/junit-platform-runner-1.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/org/junit/platform/junit-platform-launcher/1.0.0-M5/junit-platform-launcher-1.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/org/junit/platform/junit-platform-engine/1.0.0-M5/junit-platform-engine-1.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/org/junit/platform/junit-platform-suite-api/1.0.0-M5/junit-platform-suite-api-1.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/junit/junit/4.12/junit-4.12.jar  /Users/manavdeepbrar/.m2/repository/org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar  /Users/manavdeepbrar/.m2/repository/org/junit/platform/junit-platform-surefire-provider/1.0.0-M5/junit-platform-surefire-provider-1.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/org/apache/maven/surefire/common-java5/2.19.1/common-java5-2.19.1.jar  /Users/manavdeepbrar/.m2/repository/org/junit/jupiter/junit-jupiter-engine/5.0.0-M5/junit-jupiter-engine-5.0.0-M5.jar  /Users/manavdeepbrar/.m2/repository/org/apache/maven/reporting/maven-reporting-api/3.0/maven-reporting-api-3.0.jar
[DEBUG] boot(compact) classpath:  surefire-booter-2.20.jar  surefire-api-2.20.jar  surefire-logger-api-2.20.jar  test-classes  classes  junit-jupiter-api-5.0.0-M5.jar  opentest4j-1.0.0-M2.jar  junit-platform-commons-1.0.0-M5.jar  junit-jupiter-params-5.0.0-M5.jar  junit-platform-runner-1.0.0-M5.jar  junit-platform-launcher-1.0.0-M5.jar  junit-platform-engine-1.0.0-M5.jar  junit-platform-suite-api-1.0.0-M5.jar  junit-4.12.jar  hamcrest-core-1.3.jar  junit-platform-surefire-provider-1.0.0-M5.jar  common-java5-2.19.1.jar  junit-jupiter-engine-5.0.0-M5.jar  maven-reporting-api-3.0.jar
[DEBUG] Forking command line: /bin/sh -c cd /Users/manavdeepbrar/Documents/Coursera_algorithmic_toolbox/primitive.array && /Library/Java/JavaVirtualMachines/jdk1.8.0_112.jdk/Contents/Home/jre/bin/java -jar /Users/manavdeepbrar/Documents/Coursera_algorithmic_toolbox/primitive.array/target/surefire/surefirebooter2469316749007257509.jar /Users/manavdeepbrar/Documents/Coursera_algorithmic_toolbox/primitive.array/target/surefire 2017-07-16T18-55-03_793-jvmRun1 surefire8111336754767961212tmp surefire_06517069218495854312tmp
[INFO] Running com.gmail.manavbrar.primitivearray.AppTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 5.603 s
[INFO] Finished at: 2017-07-16T18:55:07-07:00
[INFO] Final Memory: 15M/1781M
[INFO] ------------------------------------------------------------------------
[ERROR] Java heap space -> [Help 1]
java.lang.OutOfMemoryError: Java heap space
	at java.util.Arrays.copyOf(Arrays.java:3332)
	at java.lang.AbstractStringBuilder.ensureCapacityInternal(AbstractStringBuilder.java:124)
	at java.lang.AbstractStringBuilder.append(AbstractStringBuilder.java:448)
	at java.lang.StringBuilder.append(StringBuilder.java:136)
	at org.apache.maven.plugin.surefire.booterclient.output.MultipleFailureException.getLocalizedMessage(MultipleFailureException.java:52)
	at org.apache.maven.plugin.surefire.booterclient.ForkStarter$CloseableCloser.run(ForkStarter.java:200)
	at org.apache.maven.surefire.shade.org.apache.maven.shared.utils.cli.CommandLineUtils$1.call(CommandLineUtils.java:282)
	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:626)
	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:533)
	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:279)
	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:243)
	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1077)
	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:907)
	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:785)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
```

sample code that will produce the error https://github.com/braghome/primitive.array

## Deliverables

- I would like to have parameters tests work with primitive array

--- a/documentation/src/docs/asciidoc/release-notes-5.0.0-M6.adoc
+++ b/documentation/src/docs/asciidoc/release-notes-5.0.0-M6.adoc
@@ -47,6 +47,8 @@ on GitHub.
 
 * All tags declared via `@Tag` are now trimmed in order to remove leading and trailing
   whitespace. Furthermore, blank tags are now ignored.
+* All primitive array types (from `boolean[]` to `short[]`) are now supported as a
+  return type of static argument providing methods for parameterized tests.
 
 ===== Deprecations and Breaking Changes
 

--- a/junit-jupiter-params/src/test/java/org/junit/jupiter/params/provider/MethodArgumentsProviderTests.java
+++ b/junit-jupiter-params/src/test/java/org/junit/jupiter/params/provider/MethodArgumentsProviderTests.java
@@ -22,6 +22,7 @@ import java.util.Optional;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.stream.Stream;
 
+import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtensionContext;
 import org.junit.platform.commons.JUnitException;
@@ -98,12 +99,105 @@ class MethodArgumentsProviderTests {
 		assertThat(exception).hasMessageContaining("required test class is not present");
 	}
 
+	@Nested
+	class PrimitiveArrays {
+
+		@Test
+		void providesArgumentsUsingBooleanArray() {
+			Stream<Object[]> arguments = provideArguments("booleanArrayProvider");
+
+			assertThat(arguments).containsExactly(array(Boolean.TRUE), array(Boolean.FALSE));
+		}
+
+		@Test
+		void providesArgumentsUsingByteArray() {
+			Stream<Object[]> arguments = provideArguments("byteArrayProvider");
+
+			assertThat(arguments).containsExactly(array((byte) 1), array(Byte.MIN_VALUE));
+		}
+
+		@Test
+		void providesArgumentsUsingCharArray() {
+			Stream<Object[]> arguments = provideArguments("charArrayProvider");
+
+			assertThat(arguments).containsExactly(array((char) 1), array(Character.MIN_VALUE));
+		}
+
+		@Test
+		void providesArgumentsUsingDoubleArray() {
+			Stream<Object[]> arguments = provideArguments("doubleArrayProvider");
+
+			assertThat(arguments).containsExactly(array(1d), array(Double.MIN_VALUE));
+		}
+
+		@Test
+		void providesArgumentsUsingFloatArray() {
+			Stream<Object[]> arguments = provideArguments("floatArrayProvider");
+
+			assertThat(arguments).containsExactly(array(1f), array(Float.MIN_VALUE));
+		}
+
+		@Test
+		void providesArgumentsUsingIntArray() {
+			Stream<Object[]> arguments = provideArguments("intArrayProvider");
+
+			assertThat(arguments).containsExactly(array(47), array(Integer.MIN_VALUE));
+		}
+
+		@Test
+		void providesArgumentsUsingLongArray() {
+			Stream<Object[]> arguments = provideArguments("longArrayProvider");
+
+			assertThat(arguments).containsExactly(array(47L), array(Long.MIN_VALUE));
+		}
+
+		@Test
+		void providesArgumentsUsingShortArray() {
+			Stream<Object[]> arguments = provideArguments("shortArrayProvider");
+
+			assertThat(arguments).containsExactly(array((short) 47), array(Short.MIN_VALUE));
+		}
+	}
+
+	@SuppressWarnings("unused")
 	static class TestCase {
 
 		static AtomicBoolean collectionStreamClosed = new AtomicBoolean(false);
 
 		static Stream<String> stringStreamProvider() {
 			return Stream.of("foo", "bar");
+		}
+
+		static boolean[] booleanArrayProvider() {
+			return new boolean[] { true, false };
+		}
+
+		static byte[] byteArrayProvider() {
+			return new byte[] { (byte) 1, Byte.MIN_VALUE };
+		}
+
+		static char[] charArrayProvider() {
+			return new char[] { (char) 1, Character.MIN_VALUE };
+		}
+
+		static double[] doubleArrayProvider() {
+			return new double[] { 1d, Double.MIN_VALUE };
+		}
+
+		static float[] floatArrayProvider() {
+			return new float[] { 1f, Float.MIN_VALUE };
+		}
+
+		static int[] intArrayProvider() {
+			return new int[] { 47, Integer.MIN_VALUE };
+		}
+
+		static long[] longArrayProvider() {
+			return new long[] { 47L, Long.MIN_VALUE };
+		}
+
+		static short[] shortArrayProvider() {
+			return new short[] { (short) 47, Short.MIN_VALUE };
 		}
 
 		static Iterable<String> stringIterableProvider() {

--- a/junit-platform-commons/src/main/java/org/junit/platform/commons/util/CollectionUtils.java
+++ b/junit-platform-commons/src/main/java/org/junit/platform/commons/util/CollectionUtils.java
@@ -17,6 +17,7 @@ import static java.util.stream.Collectors.toList;
 import static java.util.stream.StreamSupport.stream;
 import static org.junit.platform.commons.meta.API.Usage.Internal;
 
+import java.lang.reflect.Array;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
@@ -131,6 +132,18 @@ public final class CollectionUtils {
 		}
 		if (object instanceof Object[]) {
 			return Arrays.stream((Object[]) object);
+		}
+		if (object instanceof double[]) {
+			return DoubleStream.of((double[]) object).boxed();
+		}
+		if (object instanceof int[]) {
+			return IntStream.of((int[]) object).boxed();
+		}
+		if (object instanceof long[]) {
+			return LongStream.of((long[]) object).boxed();
+		}
+		if (object.getClass().isArray() && object.getClass().getComponentType().isPrimitive()) {
+			return IntStream.range(0, Array.getLength(object)).mapToObj(i -> Array.get(object, i));
 		}
 		throw new PreconditionViolationException(
 			"Cannot convert instance of " + object.getClass().getName() + " into a Stream: " + object);

--- a/platform-tests/src/test/java/org/junit/platform/commons/util/CollectionUtilsTests.java
+++ b/platform-tests/src/test/java/org/junit/platform/commons/util/CollectionUtilsTests.java
@@ -18,8 +18,11 @@ import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertSame;
 import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.DynamicTest.dynamicTest;
 import static org.junit.platform.commons.util.CollectionUtils.toUnmodifiableList;
 
+import java.lang.reflect.Array;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Iterator;
@@ -30,7 +33,9 @@ import java.util.stream.IntStream;
 import java.util.stream.LongStream;
 import java.util.stream.Stream;
 
+import org.junit.jupiter.api.DynamicTest;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestFactory;
 
 /**
  * Unit tests for {@link CollectionUtils}.
@@ -190,4 +195,36 @@ class CollectionUtilsTests {
 		assertThat(result).containsExactly("foo", "bar");
 	}
 
+	@TestFactory
+	Stream<DynamicTest> toStreamWithPrimitiveArrays() {
+		//@formatter:off
+		return Stream.of(
+				dynamicTest("boolean[]",
+						() -> toStreamWithPrimitiveArray(new boolean[] { true, false })),
+				dynamicTest("byte[]",
+						() -> toStreamWithPrimitiveArray(new byte[] { 0, Byte.MIN_VALUE, Byte.MAX_VALUE })),
+				dynamicTest("char[]",
+						() -> toStreamWithPrimitiveArray(new char[] { 0, Character.MIN_VALUE, Character.MAX_VALUE })),
+				dynamicTest("double[]",
+						() -> toStreamWithPrimitiveArray(new double[] { 0, Double.MIN_VALUE, Double.MAX_VALUE })),
+				dynamicTest("float[]",
+						() -> toStreamWithPrimitiveArray(new float[] { 0, Float.MIN_VALUE, Float.MAX_VALUE })),
+				dynamicTest("int[]",
+						() -> toStreamWithPrimitiveArray(new int[] { 0, Integer.MIN_VALUE, Integer.MAX_VALUE })),
+				dynamicTest("long[]",
+						() -> toStreamWithPrimitiveArray(new long[] { 0, Long.MIN_VALUE, Long.MAX_VALUE })),
+				dynamicTest("short[]",
+						() -> toStreamWithPrimitiveArray(new short[] { 0, Short.MIN_VALUE, Short.MAX_VALUE }))
+		);
+		//@formatter:on
+	}
+
+	private void toStreamWithPrimitiveArray(Object primitiveArray) {
+		assertTrue(primitiveArray.getClass().isArray());
+		assertTrue(primitiveArray.getClass().getComponentType().isPrimitive());
+		Object[] result = CollectionUtils.toStream(primitiveArray).toArray();
+		for (int i = 0; i < result.length; i++) {
+			assertEquals(Array.get(primitiveArray, i), result[i]);
+		}
+	}
 }

